test/puppetlabs/puppetdb/admin_test.clj

```diff
@@ -26,26 +27,84 @@
             [puppetlabs.puppetdb.testutils.services :as svc-utils]
             [puppetlabs.puppetdb.scf.storage :as scf-storage]
             [puppetlabs.puppetdb.testutils.catalog-inputs :refer [sample-input-cmds]]
-            [puppetlabs.puppetdb.time :refer [now days ago] :as time]))
+            [puppetlabs.puppetdb.time :refer [now days ago] :as time])
+  (:import
+    (java.time.format DateTimeFormatter)
+    (java.time ZonedDateTime ZoneId)))
 
 (use-fixtures :each tu/call-with-test-logging-silenced)
 
+(defn update-timestamp [resource resource-list time]
+  (let [current-time  (time/to-date-time time)
+        index (.indexOf resource-list resource)
+        ; We need to preserve the order of the reports in example/reports
+        minus (time/minus current-time (time/parse-period (str (- 10 index) "s")))
+        timestamp (time/unparse (time/formatters :date-time) minus)
+        ]
+    (-> resource
+        (assoc :timestamp timestamp)
+        (update :events (fn [events]
+                                   (mapv #(assoc % :timestamp timestamp)
+                                         events)))
+        )
+    )
+  )
+
+(defn change-report-time
```

Comment:
We should move this helper func into `testutils.clj` and then change the existing call in `services_test.clj` to use the same helper.